File: api/service/src/main/java/au/com/rainmore/centus/services/books/BookService.java
package au.com.rainmore.centus.services.books;

import au.com.rainmore.centus.domains.books.Book;
import au.com.rainmore.centus.services.books.dto.BookDto;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
public class BookService {

    private final BookRepository   bookRepository;
    private final BookDtoConverter bookDtoConverter;

    public BookService(BookRepository bookRepository, BookDtoConverter bookDtoConverter) {
        this.bookRepository = bookRepository;
        this.bookDtoConverter = bookDtoConverter;
    }

    public Page<Book> findAll(Pageable pageable) {
        return bookRepository.findAll(pageable);
    }

    @Transactional(readOnly = true)
    public Page<BookDto> findAllDto(Pageable pageable) {
        return findAll(pageable).map(bookDtoConverter::convert);
    }

    @Transactional
    public void save(Book book) {
        bookRepository.save(book);
    }
}

File: ui/src/services/book/book-service.ts
import {apiService, ApiService } from '../api/api-service';
import { Page, Pageable } from '../api/types';
import { Book } from './types';

export class BookService {

  constructor(private apiService: ApiService) {

  }

  find(pageable?: Pageable, cancelToken?): Promise<Page<Book>> {
    return this.apiService.findPage('/api/books', undefined, pageable)
  }
}

export const bookService: BookService = new BookService(apiService)